### PR TITLE
defineCustomElement

### PR DESCRIPTION
**Pull request by littlespex**
- Moved custom element definition code into utility function
- Updated gulp build script to use ES6 modules to match the `package.json` configuration.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-const livereload = require('gulp-livereload');
-
-const pkg = require('./package.json');
-const del = require('del');
-
+import commonjs from '@rollup/plugin-commonjs';
+import resolve from '@rollup/plugin-node-resolve';
+import del from 'del';
+import fs from 'fs';
+import gulp from 'gulp';
```
Comment (littlespex) [inline gulpfile.js:5]: The `package.json` file has the `"type": "module"` flag set, so I ran into compile errors when trying to run the `build` and `dev` scripts. I converted the file to ES6 style imports. Alternatively, I could revert these changes and remove `"type": "module"` from the package file.

File: src/js/media-chrome-button.js [hunk 2]
```diff
@@ -87,16 +88,13 @@ class MediaChromeButton extends MediaChromeElement {
...
   set icon(svg) {
     this.iconContainer.innerHTML = svg;
   }
 }
 
-if (!window.customElements.get('media-chrome-button')) {
-  window.customElements.define('media-chrome-button', MediaChromeButton);
-  window.MediaChromeButton = MediaChromeButton;
-}
+defineCustomElement('media-chrome-button', MediaChromeButton);
```
Comment (littlespex) [inline src/js/media-chrome-button.js:98]: This is the bulk of the changes, converting this chunk into a util function in every component file.

File: src/js/media-chrome-menu-button.js [hunk 3]
```diff
@@ -62,12 +63,9 @@ class MediaChromeMenuButton extends MediaChromeButton {
     }
   }
 
-  mediaSetCallback(media) {}
+  mediaSetCallback(media) { }
 }
 
-if (!window.customElements.get('media-chrome-menu-button')) {
-  window.customElements.define('media-chrome-menu-button', MediaChromeMenuButton);
-  window.MediaChrome = MediaChromeMenuButton;
```
Comment (littlespex) [inline src/js/media-chrome-menu-button.js:70]: This line is one of the lines that initiated the conversion. There are a few components that register `window.{COMPONENT_NAME}` incorrectly.

File: src/js/utils/defineCustomElement.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export function defineCustomElement(name, element) {
+  if (!window.customElements.get(name)) {
+    window.customElements.define(name, element);
+    window[element.name] = element;
+  }
+}
```
Comment (littlespex) [inline src/js/utils/defineCustomElement.js:6]: Here is the util function.

File: src/js/media-chrome-menu-button.js (hunk 3 above)
Comment (heff) [reply src/js/media-chrome-menu-button.js:70]: Ha! Oops.

File: gulpfile.js (hunk 1 above)
Comment (heff) [reply gulpfile.js:5]: Ah nice. I'm not running into that for some reason but seems like the right change.

**Review by heff — APPROVED** [review]
Good stuff. Thanks Casey!
